assets/src/edit-story/components/panels/sizePosition.js

```diff
@@ -156,11 +181,51 @@ function SizePositionPanel({
     [lockAspectRatio]
   );
 
-  usePresubmitHandler(
-    ({ rotationAngle: newRotationAngle }) => ({
+  usePresubmitHandler(({ rotationAngle: newRotationAngle }) => {
+    return {
       rotationAngle: newRotationAngle % 360,
-    }),
-    []
+    };
+  }, []);
+
+  const setDimensionMinMax = useCallback(
+    (value, ratio, minmax) => {
+      if (lockAspectRatio && value >= minmax.MAX) {
+        return clamp(minmax.MAX * ratio, minmax);
+      }
+
+      return clamp(value, minmax);
+    },
+    [lockAspectRatio]
+  );
+
+  usePresubmitHandler(
+    ({ height: newHeight }, { width: oldWidth, height: oldHeight }) => {
+      const ratio = oldHeight / oldWidth;
+      if (isNum(ratio)) {
+        return {
+          height: setDimensionMinMax(
+            dataPixels(newHeight),
+            ratio,
+            MIN_MAX.HEIGHT
+          ),
+        };
+      }
+      return null;
+    },
+    [height, lockAspectRatio]
+  );
+
+  usePresubmitHandler(
+    ({ width: newWidth }, { width: oldWidth, height: oldHeight }) => {
+      const ratio = oldWidth / oldHeight;
+      if (isNum(ratio)) {
+        return {
+          width: setDimensionMinMax(dataPixels(newWidth), ratio, MIN_MAX.WIDTH),
```

Comment:
Maybe set the `min` and `max` to `BoxedNumeric` below for width and height as well?

Reply:
Applied them to even more inputs https://github.com/google/web-stories-wp/pull/2255/commits/da9defe1e904a72681a3425950d67c4e91929bc7